content/articles/object-serialization-and-deserialization-in-java/index.md

```diff
@@ -0,0 +1,343 @@
+        Link linkFromFile = deserializeLink(filename);
+        System.out.println("Following Serialization : ");
+        linkFromFile.printLink();
+    }
+}
+```
+
+#### Output
+
+```
+Before Serialization : 
+Link : My first link
+After Serialization : 
+Link : My first link
+```
+
+### How to use inheritance to serialize an object in Java
+
+When we mark a parent class as serializable, all its subclasses are implicitly serializable. Thus, you don't need the subclasses to implement the Serializable interface directly.
+
+#### Example
+
+```java
+import java.io.*;
+
+class Link implements Serializable {
+
+    private String commons;
+    protected int favorites;
+
+    public Link(String commons, int favorites) {
+        this.commons = commons;
+        this.favorites = favorites;
+    }
+
+    public void printLink() {
+        System.out.println("Link : " + this.commons);
+    }
+
+}
+class SerialLink extends Link {
+    protected String commits;
+
+    public void addCommit(String newCommit)
+    {
+        if (this.commits == null) commits = "";
+        this.commits += (newCommit + ",");
+    }
+
+    public String getCommit() {
+        return this.commits;
+    }
+
+    public SerialLink(String commons, int favorites) {
+        super(commons, favorites);
+    }
+}
+```
+
+#### Description
+
+There are two classes in the example above: `Link` and `SerialLink`. `SerialLink` derives from `Link` and has a `commits` data member that contains all the commits left on the link. Due to the concept of Multilevel Inheritance, `Link` implements `Serializable`, and so `SerialLink` will become serializable as well. However, this idea does not operate in reverse as a result. When a child class implements the Serializable interface, the Parent class is not affected.
+
+### Serialization with aggregation
+
+To establish a *HAS-A* connection in Java, we utilize aggregation. This implies that a single class may refer to many different other classes. These classes won't be serializable until all of their references are of type `Serializable`. Any attempt to serialize these types fails due to `NotSerializableException`.
+
+#### Example
+
+```java
+class Link {
+    private String heading;
+    private String commons;
+    public String getlink() {
+        return this.commons + " : " + this.link;
+    }
+}
+class Seriallink {
+    private link link;
+    private int favorites;
+}
+```
+
+#### Description
+
+In the illustration above, if a client wants to serialize the `SerialLink` object, a `NotSerizalizableException` will be produced. Since `link` cannot be serialized until the `Serializable` interface has been implemented, this error will occur.
+
+When properties aren't serialized, there are a couple of things to consider:
+
+1. Static members
+2. Transient members
+
+### Serialization with static data members
+
+Any static data members of a class will not be serialized because static denotes a class property, not an object property.
+
+```java

```

Comment:
Going over this again, I was wondering how this code snippet with the `Lecturer` class is relevant to the content being discussed here. This code snippet seems out of place and I would either remove it or add an explanation about it.